components/brave_wallet_ui/common/async/hardware.ts

```diff
@@ -71,7 +71,7 @@ export async function signTrezorTransaction (
   txInfo.txDataUnion.ethTxData1559.baseData.nonce = nonce.nonce
   const signed = await deviceKeyring.signTransaction(path, txInfo, chainId.chainId)
   if (!signed || !signed.success || !signed.payload) {
-    const error = signed.error ? signed.error : getLocale('braveWalletSignOnDeviceError')
+    const error = (signed.error ? signed.error : getLocale('braveWalletSignOnDeviceError')) as string
```

Comment:
can we accomplish this without a type cast?

also shorthand for this would be `const error = signed.error  ?? getLocale('braveWalletSignOnDeviceError')`

Reply:
> I think ideally we have one common error type and have the Trezor and Ledger logic convert to that common type before this point so that no cast is needed, but I think that's out of scope for now.

That sounds like a good way to handle this to me, but yeah I wouldn't consider it an absolute necessity either